classes/Supplier.php

```diff
@@ -154,9 +154,8 @@ public static function getSuppliers($getNbProducts = false, $idLang = 0, $active
 					JOIN `' . _DB_PREFIX_ . 'product` p ON (ps.`id_product`= p.`id_product`)
 					' . Shop::addSqlAssociation('product', 'p') . '
 					LEFT JOIN `' . _DB_PREFIX_ . 'supplier` as m ON (m.`id_supplier`= p.`id_supplier`)
-					WHERE ps.id_product_attribute = 0' .
+					WHERE product_shop.`visibility` NOT IN ("none")' .
```

Comment:
Do we really need to use NOT IN? why not `<>` ?

Reply:
It's the same you're right I just copied the existing query condition, I want to limit the modifications on legacy code